### PR TITLE
feat: Add BuiltBlockTracerError

### PR DESCRIPTION
**Pull request by ghost**
## 📝 Summary

- Add BuitlBlockTracerError
- Add .code and .idea to `.gitignore`
- And lint

## 💡 Motivation and Context

---

## ✅ I have completed the following steps:

* [x] Run `make lint`
* [x] Run `make test`
* [ ] Added tests (if applicable)

### CODE REVIEW

File: crates/rbuilder/src/building/built_block_trace.rs [hunk 1]
```diff
@@ -78,24 +90,24 @@ impl BuiltBlockTrace {
             })
     }
 
-    pub fn verify_bundle_consistency(&self, blocklist: &HashSet<Address>) -> eyre::Result<()> {
+    pub fn verify_bundle_consistency(
+        &self,
+        blocklist: &HashSet<Address>,
+    ) -> eyre::Result<(), BuiltBlockTraceError> {
```
Comment (ZanCorDX) [inline crates/rbuilder/src/building/built_block_trace.rs]: We don't need eyre anymore. Replace it for std::Result.

**Comment by ZanCorDX** [comment]
Thanks for your interest in the rbuilder and being the first external contributor!
Now about the code:
- Can you rebase?
- The PR seems to also change:
   - Add .idea to gitignore (can you also add .code?)
   - Restructured some “use”
   
   So it would be nice if you put that also on the description.

**Comment by metachris** [comment]
Don't mind the benchmark comment about degraded performance. It's a bit flaky, and should be fixed by https://github.com/flashbots/rbuilder/pull/15 which we intend to merge soon.

**Comment by ghost** [comment]
> Don't mind the benchmark comment about degraded performance. It's a bit flaky, and should be fixed by #15 which we intend to merge soon.

Okay, I see.

**Comment by ZanCorDX** [comment]
Good work!
Next time, try making separate PRs for things like lint/gitignore and "real" code changes.
This time you get a free pass for being the first collaborator!

**Review by ZanCorDX — APPROVED** [review]

**Pull request merged**